homeassistant/components/starlink/coordinator.py

```diff
@@ -53,3 +54,13 @@ async def _async_update_data(self) -> StarlinkData:
                 return StarlinkData(*status)
             except GrpcError as exc:
                 raise UpdateFailed from exc
+
+    async def stow_starlink(self, stow: bool):
+        """Set whether Starlink should be stowed."""
+        async with async_timeout.timeout(4):
+            try:
+                await self.hass.async_add_executor_job(
+                    set_stow_state, not stow, self.channel_context
+                )
+            except GrpcError as exc:
+                raise UpdateFailed from exc
```

Comment:
Same deal here, what's the ideal behavior for exceptions thrown by switches and buttons?

Reply:
I think I've answered this in the other PR, but happy to clarify if needed

Reply:
Yep, both throw a HomeAssistantError now. Thanks!